src/v2/SuperstateToken.sol

```diff
@@ -0,0 +1,463 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.26;
+
+import {ERC20Upgradeable} from "openzeppelin-contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import {Ownable2StepUpgradeable} from "openzeppelin-contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {PausableUpgradeable} from "openzeppelin-contracts-upgradeable/security/PausableUpgradeable.sol";
+import {ECDSA} from "openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+
+import {console} from "forge-std/console.sol";
+import {IERC7246} from "src/interfaces/IERC7246.sol";
+import {AllowList} from "src/AllowList.sol";
+
+/**
+ * @title SuperstateToken
+ * @notice A Pausable ERC7246 token contract that interacts with the AllowList contract to check if transfers are allowed
+ * @author Superstate
+ */
+abstract contract SuperstateToken is ERC20Upgradeable, IERC7246, PausableUpgradeable, Ownable2StepUpgradeable {
```

Comment:
Mirrored this in its own directory so that we have a history of past versions, and so we can more easily test upgrading the version in unit tests.

Chose to not use contract inheritance against prior versions of the contract as I don't think that'll scale well for future upgades.